### PR TITLE
Remove glyph FRP

### LINKED ISSUE #5724: Eliminate Glyph FRP Network
The application is spending a lot of time in `new_glyph`, largely setting up FRP networks. It costs 0.8s in *Orders*, and scales up linearly with the amount of renderable data. Let's eliminate per-glyph FRP networks.

>     1. FRP `set_color` can be changed to just a function.
> 
>     2. FRP `target_color` can be converted to variable in glyph.
> 
>     3. The color animation can be removed from glyphs (sure, worse effect, but big improvement of removing FRP).
> 
>     4. The FRP network with the title `=== Debug mode ===` is used for debug purposes only. Let's remove it for now. We can code it different way around in our debug scene.

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Implements #5724. Cuts `new_glyph` time in half. I'm looking in to the remainder of the time...

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

**Comment by kazcw** [comment]
Most of the remaining time in `new_glyph` is spent creating the display::Object and instantiating the shape system:

|self_duration|total_duration|count|profiler                                                                    |
|-------------|--------------|-----|----------------------------------------------------------------------------|
| 220.7       | 220.7        |599  |instantiate (lib/rust/ensogl/core/src/display/shape/primitive/system.rs:303)|
| 107.2       | 107.2        |293  |new (lib/rust/ensogl/core/src/display/object/instance.rs:1122)              |
|  73.0       | 401.3        |1107 |new_glyph (lib/rust/ensogl/component/text/src/font/glyph.rs:552)            |
|   0.4       |   0.4        |1    |new (lib/rust/ensogl/core/src/display/shape/primitive/system.rs:272)        |

These are worth looking in to after more impactful planned improvements like #5722.

**Comment by wdanilo** [comment]
@kazcw , what about updating the text-area debug scene? Have you tried it? After removing these FRP bindings here, the scene does not allow for testing bigger glyphs, different colors, etc. It would be cool to allow for it by implementing the removed logic there. Have you tried doing it?

**Comment by kazcw** [comment]
> @kazcw , what about updating the text-area debug scene? Have you tried it? After removing these FRP bindings here, the scene does not allow for testing bigger glyphs, different colors, etc. It would be cool to allow for it by implementing the removed logic there. Have you tried doing it?

Implemented. The bindings to change hinting are now Ctrl+(Shift)+E to increase (decrease) `opacity_exponent`, and Ctrl+(Shift)+O to increase (decrease) `opacity_increment`. Other bindings are unchanged. Internally, hinting parameters are now maintained per font, not per glyph.

File: lib/rust/ensogl/component/text/src/font.rs [hunk 1]
```diff
@@ -842,3 +846,59 @@ impl scene::Extension for Registry {
...
+            (Some(platform::Platform::MacOS), "mplus1p") =>
+                Self { opacity_increase: 0.4, opacity_exponent: 4.0 },
+            (Some(platform::Platform::Windows), "mplus1p") =>
+                Self { opacity_increase: 0.3, opacity_exponent: 3.0 },
+            (Some(platform::Platform::Linux), "mplus1p") =>
+                Self { opacity_increase: 0.3, opacity_exponent: 3.0 },
+            _ => Self::default(),
+        }
+    }
+}
```
Comment (farmaazon) [inline lib/rust/ensogl/component/text/src/font.rs]: I think we should add explanation here why do we need it and why it differs.

File: lib/rust/ensogl/component/text/src/font/glyph.rs [hunk 2]
```diff
@@ -526,7 +461,8 @@ impl WeakGlyph {
 // ==============
 
 #[cfg(not(target_arch = "wasm32"))]
-#[derive(Clone, Copy, CloneRef, Debug, Default)]
+#[derive(Clone, CloneRef, Debug, Default)]
+#[allow(missing_copy_implementations)]
```
Comment (farmaazon) [inline lib/rust/ensogl/component/text/src/font/glyph.rs:465]: Why cannot we implement `Copy`?

File: lib/rust/profiler/data/src/aggregate.rs [hunk 3]
```diff
@@ -21,8 +40,15 @@ pub struct Aggregator {
...
-            self.visit_interval(profile, *child);
+        for &child in &profile.root_interval().children {
+            let interval = &profile[child];
+            let measurement = &profile[interval.measurement];
+            if measurement.classify() == Class::OnFrame
+                && interval.interval.duration_ms().unwrap_or_default() < SKIP_FRAMES_BELOW_MS
+            {
+                continue;
+            }
+            self.visit_interval(profile, child);
```
Comment (farmaazon) [inline lib/rust/profiler/data/src/aggregate.rs]: I prefer iterating over `profile.root_interval().children.iter().filter(...)` than `continue` jumps.

File: lib/rust/profiler/data/src/bin/intervals.rs [hunk 4]
```diff
@@ -92,13 +102,19 @@ impl FuncCollector {
...
-    fn visit(&mut self, label: &Label, frame: &data::aggregate::Frame) {
-        let func = self.funcs.entry(label.clone()).or_default();
-        func.self_duration += frame.self_duration();
-        func.total_duration += frame.total_duration();
-        func.count += frame.interval_count();
+    fn visit(&mut self, label: &Label, frame: &data::aggregate::Frame, mut enable: bool) {
+        if let Some(prefix) = INCLUDE_ONLY_SUBTREES_MATCHING_PREFIX
+                && label.starts_with(prefix) {
+            enable = true;
+        }
```
Comment (farmaazon) [inline lib/rust/profiler/data/src/bin/intervals.rs]:
```rust
let enabled = enable || INCLUDE_ONLY_SUBTREES_MATCHING_PREFIX.map_or(false, |pfx| label.starts_with(pfx));
if enabled {...
```
I feel it shows the logic more clearly.

**Review by farmaazon — APPROVED** [review]

File: lib/rust/ensogl/component/text/src/font/glyph.rs (hunk 2 above)
Comment (kazcw) [reply lib/rust/ensogl/component/text/src/font/glyph.rs:465]: This is a mock of a non-`Copy` type; if it's `Copy`, the linter will complain when we clone it.

Comment (farmaazon) [reply lib/rust/ensogl/component/text/src/font/glyph.rs:465]: Ok. You can add a comment there, so nobody will ask this question again.

**Review by wdanilo — APPROVED** [review]

**Pull request merged**